### PR TITLE
Fixed date comparison error in scrape_inmates logic

### PR DESCRIPTION
**Pull request by nwinklareth**

### CODE REVIEW

**Comment by bepetersn** [comment]
Hey @nwinklareth, did you see my clone_db script that I submitted a pull
request for? It's fully functional.

**Comment by nwinklareth** [comment]
Yes Brian I did and I did merge it in.

Nice piece of meta-programming. Now we need to figure out how to solve the
timeout problem.

Norbert

On Sun, Aug 4, 2013 at 1:15 AM, Brian Peterson notifications@github.comwrote:

> Hey @nwinklareth, did you see my clone_db script that I submitted a pull
> request for? It's fully functional.
> On Aug 3, 2013 4:34 PM, "nwinklareth" notifications@github.com wrote:
> 
> > ---
> > 
> > You can merge this Pull Request by running
> > 
> > git pull https://github.com/nwinklareth/cookcountyjail master
> > 
> > Or view, comment on, or merge it at:
> > 
> > https://github.com/sc3/cookcountyjail/pull/98
> > Commit Summary
> > - Fixed date comparison error in scrape_inmates logic
> > 
> > File Changes
> > - _M_ cookcountyjail/settings.py<
> >   https://github.com/sc3/cookcountyjail/pull/98/files#diff-0>(6)
> > - _M_ countyapi/management/commands/scrape_inmates.py<
> >   https://github.com/sc3/cookcountyjail/pull/98/files#diff-1>(21)
> > 
> > Patch Links:
> > - https://github.com/sc3/cookcountyjail/pull/98.patch
> > - https://github.com/sc3/cookcountyjail/pull/98.diff
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/sc3/cookcountyjail/pull/98#issuecomment-22067417
> .

## 

Regards

## Norbert

Norbert Winklareth

**Comment by bepetersn** [comment]
I saw that Norbert, thanks. I think I could make it an 'update' rather than
'clone ' DB pretty easily.

**Comment by nwinklareth** [comment]
Your welcome.

Great and we need to fix the time out problem first. The only viable long
term solution that I can think of is for db_clone to fetch in chunks, say
25,000 records at a time.

Would you like to take that on? By the way figuring out how to do paging
will help in other areas.

Norbert

On Sun, Aug 4, 2013 at 6:00 PM, Brian Peterson notifications@github.comwrote:

> I saw that Norbert, thanks. I think I could make it an 'update' rather
> than
> 'clone ' DB pretty easily.
> On Aug 4, 2013 5:45 PM, "nwinklareth" notifications@github.com wrote:
> 
> > Yes Brian I did and I did merge it in.
> > 
> > Nice piece of meta-programming. Now we need to figure out how to solve
> > the
> > timeout problem.
> > 
> > Norbert
> > 
> > On Sun, Aug 4, 2013 at 1:15 AM, Brian Peterson notifications@github.comwrote:
> > 
> > > Hey @nwinklareth, did you see my clone_db script that I submitted a
> > > pull
> > > request for? It's fully functional.
> > > On Aug 3, 2013 4:34 PM, "nwinklareth" notifications@github.com
> > > wrote:
> > > 
> > > > ---
> > > > 
> > > > You can merge this Pull Request by running
> > > > 
> > > > git pull https://github.com/nwinklareth/cookcountyjail master
> > > > 
> > > > Or view, comment on, or merge it at:
> > > > 
> > > > https://github.com/sc3/cookcountyjail/pull/98
> > > > Commit Summary
> > > > - Fixed date comparison error in scrape_inmates logic
> > > > 
> > > > File Changes
> > > > - _M_ cookcountyjail/settings.py<
> > > >   https://github.com/sc3/cookcountyjail/pull/98/files#diff-0>(6)
> > > > - _M_ countyapi/management/commands/scrape_inmates.py<
> > > >   https://github.com/sc3/cookcountyjail/pull/98/files#diff-1>(21)
> > > > 
> > > > Patch Links:
> > > > - https://github.com/sc3/cookcountyjail/pull/98.patch
> > > > - https://github.com/sc3/cookcountyjail/pull/98.diff
> > > 
> > > —
> > > Reply to this email directly or view it on GitHub<
> > > https://github.com/sc3/cookcountyjail/pull/98#issuecomment-22067417>
> > > .
> > 
> > ## 
> > 
> > Regards
> > 
> > ## Norbert
> > 
> > Norbert Winklareth
> > 
> > —
> > Reply to this email directly or view it on GitHub<
> > https://github.com/sc3/cookcountyjail/pull/98#issuecomment-22080642>
> > .
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/sc3/cookcountyjail/pull/98#issuecomment-22080881
> .

## 

Regards

## Norbert

Norbert Winklareth

File: countyapi/management/commands/scrape_inmates.py [hunk 1]
```diff
@@ -109,14 +109,23 @@ def calculate_discharge_date(self, seen):
...
             booking_year = datetime.strptime(booking_date[0:4], '%Y')
             if booking_year.year < start_date.year:
                 return True
-            booking_month_date = datetime.strptime(booking_date[5:9], '%m%d')
-            return booking_month_date.month <= start_date.month and booking_month_date.day < start_date.day
+            the_month = booking_date[5:7]
+            the_day = booking_date[7:9]
+            inmates_booking_date = date(booking_year.year, convert_to_int(the_month, 1),
+                                        convert_to_int(the_day, 1))
+            if href == 'details.asp?jailnumber=2013-0731005':
```
Comment (eads) [inline countyapi/management/commands/scrape_inmates.py:126]: Yikes, what's happening here?

Comment (bepetersn) [inline countyapi/management/commands/scrape_inmates.py:126]: It looks like some debugging printouts that shouldn't have stayed in, to my eyes. Unless we need to look at a particular inmate's details each scrape.

Comment (nwinklareth) [inline countyapi/management/commands/scrape_inmates.py:126]: Whoops sorry about that I had a problem and needed the test to see when I
finally had fixed the problem.Rip out the guard and the text under it.

On Thu, Aug 8, 2013 at 2:45 AM, Brian Peterson notifications@github.comwrote:

> In countyapi/management/commands/scrape_inmates.py:
> 
> > ```
> >          # Also prior to 2010 the format of a jail id is not in the format YYYY-MMDDNNN it was YYYY-NNNNNNN
> >          # so have to parse for year first and is before this one
> >          booking_date = href.split('=')[1]
> >          booking_year = datetime.strptime(booking_date[0:4], '%Y')
> >          if booking_year.year < start_date.year:
> >              return True
> > ```
> > -            booking_month_date = datetime.strptime(booking_date[5:9], '%m%d')
> > -            return booking_month_date.month <= start_date.month and booking_month_date.day < start_date.day
> > -            the_month = booking_date[5:7]
> > -            the_day = booking_date[7:9]
> > -            inmates_booking_date = date(booking_year.year, convert_to_int(the_month, 1),
> > -                                        convert_to_int(the_day, 1))
> > -            if href == 'details.asp?jailnumber=2013-0731005':
> 
> It looks like some debugging printouts that shouldn't have stayed in, to
> my eyes. Unless we need to look at a particular inmate's details each
> scrape.
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/sc3/cookcountyjail/pull/98/files#r5652095
> .

## 

Regards

## Norbert

Norbert Winklareth

**Pull request merged**